packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -0,0 +1,350 @@
+ * @param nVariations Scalar number of variations.
+ * @param alpha false positive rate (default: 0.05).
+ * @param twoTailed Binary indicator if the test is 1 or 2-tailed (default: true).
+ * @returns Estimated power.
+ */
+export function powerEst(
+  effectSize: number,
+  mean: number,
+  variance: number,
+  n: number,
+  nVariations: number,
+  alpha: number = 0.05,
+  twoTailed: boolean = true,
+  sequentialTuningParameter = 0,
+): number {
+  if (typeof twoTailed !== "boolean") {
+    throw new Error("twoTailed must be boolean.");
+  }
+  const zStar = twoTailed
+    ? normal.quantile(1.0 - 0.5 * alpha, 0, 1)
+    : normal.quantile(1.0 - alpha, 0, 1);
+
+  let standardError = 0;
+  if (sequentialTuningParameter > 0) {
+    standardError = sequentialPowerStandardError(
+      variance,
+      mean,
+      n,
+      nVariations,
+      effectSize,
+      alpha,
+      sequentialTuningParameter,
+    );
+  } else {
+    standardError = powerStandardError(
+      variance,
+      mean,
+      n,
+      nVariations,
+      effectSize,
+    );
+  }
+  const standardizedEffectSize = effectSize / standardError;
+  const upperCutpoint = zStar - standardizedEffectSize;
+  let power = 1 - normal.cdf(upperCutpoint, 0, 1);
+  if (twoTailed) {
+    const lowerCutpoint = -zStar - standardizedEffectSize;
+    power += normal.cdf(lowerCutpoint, 0, 1);
+  }
+  return power;
+}
+
+/**
+ * Calculates minimum detectable effect
+ *
+ * @param power desired power.
+ * @param mean Scalar mean of the distribution.
+ * @param variance Scalar variance of the distribution.
+ * @param n Scalar sample size.
+ * @param nVariations Scalar number of variations.
+ * @param alpha false positive rate (default: 0.05).
+ * @returns Estimated power.
+ */
+export function findMde(
+  power: number,
+  mean: number,
+  variance: number,
+  n: number,
+  nVariations: number,
+  alpha: number = 0.05,
+  sequentialTuningParameter = 0,
+): MDEResults {
+  // Error handling:
+  if (power <= alpha) {
+    const mdeResults: MDEResults = {
+      type: "error",
+      description: "power must be greater than alpha.",
+    };
+    return mdeResults;
+  }
+  const nA = n / nVariations;
+  const z =
+    normal.quantile(1.0 - 0.5 * alpha, 0, 1) -
+    normal.quantile(1.0 - power, 0, 1);
+  let v = variance;
+  if (sequentialTuningParameter > 0) {
+    v = sequentialPowerSequentialVariance(
+      variance,
+      nA,
+      alpha,
+      sequentialTuningParameter,
+    );
+  }
+  if (nA <= (v * z ** 2) / (2 * mean ** 2)) {
+    const mdeResults: MDEResults = {
+      type: "error",
+      description:
+        "need to increase number of users or reduce number of variations.",
+    };
+    return mdeResults;
+  }
+  const sigma2 = v / nA;
+  const a_star = 1 - (z ** 2 * sigma2) / mean ** 2;
+  const b_star = -2 * mean;
+  const c_star = mean ** 2 - z ** 2 * sigma2;
+  const disc = b_star ** 2 - 4 * a_star * c_star;
+  const sol_1 = (-b_star + Math.sqrt(disc)) / (2 * a_star);
```

Comment:
fwiw it's a little hard for me to evaluate this without the documentation on what it's doing. I don't really see the mapping from this to the Power notion doc.

Reply:
agreed this is hard to evaluate, this is the quadratic form from wolfram, i'll push documentation to clarify.